doc/tools/coqrst/coqdomain.py

```diff
@@ -1070,6 +1069,16 @@ class CoqSubdomainsIndex(Index):
     name, localname, shortname, subdomains = None, None, None, [] # Must be overwritten
 
     def generate(self, docnames=None):
+        if self.name == 'glossindex':
+            # copy index data from std domain
+            env = CoqDomain.env
```

Comment:
I'm not sure about capturing env.  We have to do this because `self.domain` is the wrong domain (`coq` instead of `std`), but maybe it would be cleaner to put this index in the std domain, and then we wouldn't have to capture env?  IIRC you can use `app.add_index_to_domain('std', GlossaryIndex)` and you'll get the right domain passed to you.